prune_shard_list.c

```diff
@@ -39,10 +39,20 @@
 #include "utils/errcodes.h"
 #include "utils/lsyscache.h"
 #include "utils/typcache.h"
+#include "utils/memutils.h"
+
+
+/*
+ * OperatorTypeCache is used for caching base operator types for given typeId,
```

Comment:
`OperatorType` doesn't make sense as a phrase. It could make sense to say we're caching the operator, or we could be more direct and say we're caching the operator identifier. `OperatorIdCache` makes the most sense.

Update the comment too: not sure what _base_ means in this context. Maybe just say _is used for caching operator identifiers for a given_.